include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -851,24 +853,37 @@ class alignment_file_output_format_REMOVEME<format_bam> : alignment_file_output_
             // ---------------------------------------------------------------------
             // Writing the Record
             // ---------------------------------------------------------------------
+            int32_t ref_length{};
 
-            // compute possible distance from alignment end to sequence end
-            // which indicates soft clipping at the end.
-            // This should be replaced by a free count_gaps function for
-            // aligned sequences which is more efficient if possible.
-            auto off_end{std::ranges::distance(seq) - offset};
+            // if alignment is non-empty, replace cigar_vector.
+            // else, compute the ref_length from given cigar_vector which is needed to fill field `bin`.
+            if (!std::ranges::empty(get<0>(align)) && !std::ranges::empty(get<1>(align)))
+            {
+                ref_length = std::ranges::distance(get<1>(align));
 
-            for (auto chr : get<1>(align))
-                if (chr == gap{})
-                    ++off_end;
+                // compute possible distance from alignment end to sequence end
+                // which indicates soft clipping at the end.
+                // This should be replaced by a free count_gaps function for
+                // aligned sequences which is more efficient if possible.
+                int32_t off_end{static_cast<int32_t>(std::ranges::distance(seq)) - offset};
 
-            off_end -= std::ranges::distance(get<1>(align));
+                for (auto chr : get<1>(align))
+                    if (chr == gap{})
```

Comment:
`off_end += static_cast<uint8_t>(chr == gap{});`
Actually, I checked on compiler explorer and both versions seem to result in the same code. Still it would be good to be explicit

Reply:
isn't the if check more explicit?

Reply:
not explicit for the human eye I mean, but for the compiler 🤖

Reply:
If it is the same code I vote for more explicit to the eye :eye:

Reply:
ok, leave it,